### DISCUSSION TITLE
Config.sh and Run.sh return "Must not run with sudo"

### DISCUSSION THREAD

**Discussion by montanizstills** [community/community#52343] [Actions]
### Self-Hosted Runner Config Script

Bug

### Body

I am not sure if this is a bug or purposeful but the config.sh and run.sh scripts have a clause where any user with the `RUNER_ALLOW_AS_ROOT` flag turned on cannot execute them. Why? Inspecting the svc.sh script it will run as a root-like user. But it must be first configured with the config script which does not like root-like users. Can someone explain this logic?

**Comment by Yarden-zamir** [comment]
Maybe 
```
RUNNER_ALLOW_RUNASROOT
```
instead of 
```
RUNER_ALLOW_AS_ROOT
```

**Reply by montanizstills** [reply]
Yes, I have a typo in my issue, but the problem remains the same. Git must fix it. An alternative would be to create a user and transfer ownership of files, but the action file will run as a 'sudoer' thus defeating the purpose.

**Reply by Yarden-zamir** [reply]
I just tested it myself and the config.sh works as expected. I ran
```bash
sudo RUNNER_ALLOW_RUNASROOT=true ./config.sh
```
which in your case will be
```bash
RUNNER_ALLOW_RUNASROOT=true ./config.sh
```
And the `Must not run with sudo` message did not appear and the script poceeded correctly.

if we go into the script we can see the check itself
```bash
#line 6
if [ $user_id -eq 0 -a -z "$RUNNER_ALLOW_RUNASROOT" ]; then
    echo "Must not run with sudo"
    exit 1
fi
```
This conditional expression is checking two conditions:

The value of the variable $user_id is equal to 0. (meaning root user)
The environment variable $RUNNER_ALLOW_RUNASROOT is an empty string.

If following it sill doesn't work, please provide more details on what you are running and the steps you took along the way.

**Reply by mdhasan76** [reply]
> I just tested it myself and the config.sh works as expected. I ran
> 
> ```shell
> sudo RUNNER_ALLOW_RUNASROOT=true ./config.sh
> ```
> 
> which in your case will be
> 
> ```shell
> RUNNER_ALLOW_RUNASROOT=true ./config.sh
> ```
> 
> And the `Must not run with sudo` message did not appear and the script poceeded correctly.
> 
> if we go into the script we can see the check itself
> 
> ```shell
> #line 6
> if [ $user_id -eq 0 -a -z "$RUNNER_ALLOW_RUNASROOT" ]; then
>     echo "Must not run with sudo"
>     exit 1
> fi
> ```
> 
> This conditional expression is checking two conditions:
> 
> The value of the variable $user_id is equal to 0. (meaning root user) The environment variable $RUNNER_ALLOW_RUNASROOT is an empty string.
> 
> If following it sill doesn't work, please provide more details on what you are running and the steps you took along the way.

Thanks it works fine

**Reply by elvinsadiq** [reply]
I ran sudo RUNNER_ALLOW_RUNASROOT=true ./config.sh command. I've got asked about the URL of you repository and runner register token. After I entered runner register token I got 

Http response code: NotFound from 'POST https://api.github.com/actions/runner-registration' (Request Id: xxxx:xxxx:xxxx:xxxx:xxxx)
{"message":"Not Found","documentation_url":"https://docs.github.com/rest","status":"404"}
Response status code does not indicate success: 404 (Not Found).

How can I solve this problem ?

**Comment by JPBFRCA4** [comment]
bug

**Comment by Jaspper132** [comment]
Use the following command to run the runner: 
```
sudo RUNNER_ALLOW_RUNASROOT=true ./run.sh
```

**Comment by wasit-shafi** [comment]
Does anyone found a way to permanently set the flag RUNNER_ALLOW_RUNASROOT, It is very disturbing to set it each time while running run.sh script (FYI: I don't have to run it as a background service)?
